### ISSUE TITLE
Issue with "undefined method `order' for nil:NilClass"

### ISSUE THREAD

**Issue by kun391** [JSONAPI-Resources/jsonapi-resources#619]
Hi everyone,

I following the guide [https://github.com/cerebris/peeps#initial-steps-to-create-this-app](url)
and run server with request `GET /pages` but it's show to me this error.

![no_method](https://cloud.githubusercontent.com/assets/5398707/13099825/0fd6f0c6-d569-11e5-8052-6c3cda044d39.PNG)

How to fix this one?

Thanks

**Comment by lgebhardt** [comment]
Did you override the `records` method?

**Comment by kun391** [comment]
hi @lgebhardt:
No, I'm just integrate jsonapi on my rails engine.
This is my source

```
# resources/page_resource.rb
module StaticPages
  class PageResource < BaseResource
    attributes :title, :slug, :content, :created_at
    # paginator :offset
  end
end

# controllers/application_controller.rb
module StaticPages
  class ApplicationController < ActionController::Base
    include JSONAPI::ActsAsResourceController
    protect_from_forgery with: :null_session
  end
end

# controllers/pages_controllers
require_dependency 'static_pages/application_controller'
module StaticPages
  class PagesController < ApplicationController
  end
end

# config/routes.rb
StaticPages::Engine.routes.draw do
  jsonapi_resources :pages
end
```

I run code and have error #619

**Comment by lgebhardt** [comment]
I'm not sure. Running in an engine hasn't been tested as much as without. I'd try testing it not in an engine and see if it works before moving into an engine.

**Comment by harisadam** [comment]
@kun391 I had the same issue and I could solve this by defining the `model_name` manually:

```
module YourEngine
  class Api::Admin::PostResource < JSONAPI::Resource
    model_name 'YourEngine::Post'

    attributes :title
  end
end
```

**Comment by kun391** [comment]
Thanks @harisadam. I have resolved the issue #619. :) :+1:
